pkg/query/querier.go

```diff
@@ -27,6 +27,18 @@ import (
 	"github.com/thanos-io/thanos/pkg/tracing"
 )
 
+var promqlFuncRequiresTwoSamples = map[string]struct{}{
+	"rate":                         {},
```

Comment:
Could we add xrate/xincrease/xdelta here too?

Reply:
Dont they also work with one sample though?

Reply:
Very good point :+1: